components/arm/universalrobots/ur.go

```diff
@@ -440,3 +466,84 @@ func reader(ctx context.Context, conn io.Reader, ua *URArm, onHaveData func()) e
 		}
 	}
 }
+
+const errBuiltinKinematics = "cannot use UR builtin kinematics with obstacles or interaction spaces"
+
+func (ua *URArm) useBuiltinKinematics(worldState *commonpb.WorldState, extra map[string]interface{}) (bool, error) {
+	// function to error out if trying to use world state with builtin kinematics
+	checkWorldState := func(usingBuiltinKinematics bool) (bool, error) {
+		if usingBuiltinKinematics && worldState != nil && (len(worldState.Obstacles) != 0 || len(worldState.InteractionSpaces) != 0) {
+			return false, errors.New(errBuiltinKinematics)
+		}
+		return usingBuiltinKinematics, nil
+	}
+
+	// if runtime preference is specified, obey that
+	if extra != nil {
+		if usingAtRuntime, ok := extra["builtin_kinematics"].(bool); ok {
+			return checkWorldState(usingAtRuntime)
+		}
```

Comment:
This needs an `else` to return `false, nil` if `extra["builtin_kinematics"] = false` was passed

Reply:
No, I think if you reread the logic you will see this is not necessary. In any case I am testing for this in a unit function and getting the expected results

Reply:
Ah, I see; you're right, sorry about that.